File: decompiled/sources/com/streamax/config/network/BaseListener.java
package com.streamax.config.network;

import com.streamax.config.base.BaseFragment;

public interface BaseListener {

    public interface GetListener extends BaseListener {
        void getFailure();

        void getSuccess(String str);

        String requestForGetConfig();
    }

    public interface GetStorageInfoListener extends BaseListener {
        void getFailure();

        void getSuccess(String str);
    }

    public interface RebootListener extends BaseListener {
        void rebootFailure();

        void rebootSuccess();
    }

    public interface SetListener extends BaseListener {
        String requestForSetConfig();

        void setFailure();

        void setSuccess();
    }

    BaseFragment getCurFragment();
}

File: decompiled/sources/com/streamax/config/network/BaseListener.kt
package com.streamax.config.network

import com.streamax.config.base.BaseFragment

interface BaseListener {
    fun getCurFragment(): BaseFragment

    interface GetListener : BaseListener {
        fun getFailure()

        fun getSuccess(config: String)

        fun requestForGetConfig(): String
    }

    interface GetStorageInfoListener : BaseListener {
        fun getFailure()

        fun getSuccess(storageInfo: String)
    }

    interface RebootListener : BaseListener {
        fun rebootFailure()

        fun rebootSuccess()
    }

    interface SetListener : BaseListener {
        fun requestForSetConfig(): String

        fun setFailure()

        fun setSuccess()
    }
}
